examples/vqe-examples/general_scheduler.py

```diff
@@ -0,0 +1,175 @@
+import abc
+from copy import deepcopy
+from typing import Dict, List, Tuple
+
+class HardwareConfig:
+
+    def __init__(self, qubits_per_qpu: List[int]) -> None:
+        self.qubits_per_qpu = qubits_per_qpu
+
+    @property
+    def num_qpus(self) -> int:
+        return len(self.qubits_per_qpu)
+
+    def print_configuration(self) -> None:
+        print("# Hardware Configuration #")
+        print("N QPUs: {}, qubits per qpu: {}".format(
+            self.num_qpus, self.qubits_per_qpu
+        ))
+
+class Schedule:
+    """
+    Represents a distribution schedule. One can run over the
+    distributed system to obtain an estimate to the desired expectation value.
+    
+    """
+    
+    def __init__(self, num_likelihoods: int,
+                 hardware_config: HardwareConfig, 
+                 oracle_size: int,
+                 allow_distributed: bool) -> None:
+        """
+        Parameters
+        ----------
+        num_likelihoods : int
+        hardware_config : HardwareConfig
+        allow_distributed : bool
+            True if distributed computing is allowed, False otherwise.
+            Distributed computing requires CCNOT gates for entanglement,
+            Qiskit does not currently (easily) support it.
+        """
+        self.num_likelihoods = num_likelihoods
+        self._oracle_size = oracle_size
```

Comment:
The naming convention from Andrea's version was specifically for QFT free QAE.. maybe we change `num_likelihoods` -> `num _circuits` and `_oracle_size` to `_circuit_width` to be more general.

Reply:
done

Reply:
Were the changes committed? I still see `num_likelihoods` and `_oracle_size`.